src/ducks/recurrence/api.js

```diff
@@ -3,14 +3,42 @@ import flatten from 'lodash/flatten'
 import omit from 'lodash/omit'
 import { dehydrate } from 'cozy-client'
 import maxBy from 'lodash/maxBy'
+import get from 'lodash/get'
+import groupBy from 'lodash/groupBy'
 import { getAutomaticLabelFromBundle } from './utils'
-import { queryRecurrenceTransactions } from './queries'
+import {
+  queryRecurrenceTransactions,
+  queryRecurrencesTransactions
+} from './queries'
 import { TRANSACTION_DOCTYPE, RECURRENCE_DOCTYPE } from 'doctypes'
 
 const addRelationship = (doc, relationshipName, definition) => {
   return set(doc, ['relationships', relationshipName], { data: definition })
 }
 
+const getRecurrenceIdFromRawTransaction = tr =>
+  get(tr, 'relationships.recurrence.data._id', null)
+
+/**
+ * Fetch bundles and associated transactions from CouchDB
+ * Returns "hydrated" bundles with an `ops` field containing all its transactions
+ *
+ * @param  {CozyClient} client
+ * @return {Promise<HydratedRecurrences>}
+ */
+export const fetchHydratedBundles = async client => {
+  const recurrenceCol = client.collection(RECURRENCE_DOCTYPE)
+  const { data: recurrences } = await recurrenceCol.all()
+  const { data: transactions } = await client.query(
+    queryRecurrencesTransactions(recurrences)
+  )
```

Comment:
No more than 100 transactions max per recurrence?

Reply:
It is possible but will not be frequent. I'm adding a TODO on the recurrence/api file to remember to handle this at some point, but I don't think it needs to be dealt with now.

Furthermore transactions are fetched sorted by date, and users will likely not be interested in transactions that are very old (as a reminder, to reach 100 transactions on a monthly recurrence, you have to have 9 year of history).